static/js/giphy.js

```diff
@@ -105,7 +105,7 @@ function renderGIPHYGrid(targetEl) {
                     }
 
                     compose_ui.insert_syntax_and_focus(
-                        `[](${props.images.downsized_medium.url})`,
+                        `[](${props.images.downsized_medium.url}&zulip_giphy_still_image=${props.images.fixed_width_still.url})`,
```

Comment:
What does a sample image of this form look like?

This may be something worth talking a bit with @andersk about ideas for how to encode this information better.

Reply:
Sticking our own junk onto the end of their URL seems like the wrong solution. (Especially junk that isn’t correctly escaped, but that’s a side point.)

We could grab `props.id` (3oKIPnAiaMCws8nOsE) and build the URL ourselves:

https://i.giphy.com/media/3oKIPnAiaMCws8nOsE/giphy.gif

so we only need to store one URL that we can recognize and parse later, and we can generate any other URLs we need:

https://i.giphy.com/media/3oKIPnAiaMCws8nOsE/giphy_s.gif

Reply:
Yeah, thanks!

Reply:
For this approach, would be want to do it as a `<giphy id="">` type of element that then gets markdown-processed into the image tag (with a `giphy` class and this ID attached for doing formatting in `rendered_markdown.js`)?

That model would also allow us to have a link in the corner (or similar) for title/attribution details.